Clarify when it's optional to specify the Azure subscription ID

cnotin opened the pull request:
Auth to Entra ID, with this action, does not necessarily need an Azure subscription. For example, if the goal is to access Entra ID / MS Graph only, and not the Azure RM API, it's totally unnecessary.

Related PR in doc: https://github.com/MicrosoftDocs/entra-docs/pull/983

MoChilia commented:
Thank you for your contribution! I appreciate your effort and understand your concern, but the statement might not be proper. We may need to refine this section with additional consideration.

cnotin commented:
Hum what is your concern? How it’s written or what it means? Because I’m very certain that you can login without specifying a subscription if you just want to interact with Entra ID but not Azure resources.

MoChilia commented:
The `subscription-id` is already marked as an optional parameter, 
![image](https://github.com/user-attachments/assets/a12bc638-f160-4ac2-b2c7-c1c0c763ac14)
And a detailed explanation of the `allow-no-subscriptions` parameter with an example is provided in the documentation, https://github.com/Azure/login?tab=readme-ov-file#allow-no-subscriptions.
It is unclear why additional statements are needed, and the use of 'especially for managed identity' seems inappropriate in this context.

MoChilia commented:
But I like the change about the last part 'Capability has been added to support access to **Entra** tenants without **Azure** subscriptions for both OIDC and non-OIDC. This can be useful to run **Microsoft Entra** tenant level commands, such as `az ad`. The action accepts an optional parameter `allow-no-subscriptions` which is `false` by default.'
Please update the PR if you agree, and we can proceed with merging the change into the document.

cnotin commented:
> the use of 'especially for managed identity' seems inappropriate in this context.

I changed "and" to "especially" because MIs aren't the only use-case for OIDC. For example, one can authenticate with OIDC with a federated identity bound to a service principal which isn't necessarily an MI

> Before you use Azure Login Action with OIDC, you need to configure a federated identity credential on a service principal or a managed identity. ([source](https://github.com/Azure/login#login-with-openid-connect-oidc-recommended))


Also,
> It is unclear why additional statements are needed

I've added some references to "Entra ID" here and there for folks who want to use this to login to Entra ID only, and who may be confused or discouraged by the references to Azure only.
